lib/log.rs

```diff
@@ -0,0 +1,74 @@
+use std::sync;
+use std::collections;
+use std::ffi;
+
+type log_callback_t = Box<dyn Fn(log_log_level_t, &str) + Send + Sync>;
+
+lazy_static! {
+    /* Logger configuration for each module consists of the maximal enabled
+     * log level (messages above this level are ignored) and callback.
+     */
+    static ref LOG_CONFIG: sync::RwLock<collections::HashMap<log_module_t, (log_callback_t, log_log_level_t)>> = {
+        sync::RwLock::new(collections::HashMap::new())
+    };
```

Comment:
FYI it compiles fine w/o the braces. I feel it's slightly more readable that way

Reply:
Thanks! I actually like braces better, but I did make other changes you suggested.